Add support for row CSS style

adrianbadea opened the pull request:
Implement CSS style for each row in grid.
This is very useful to dynamically apply a row style, depending on the data.
See rowStyle sample provided.

adrianbadea committed:
Add support for row CSS style

Implement CSS style for each row in grid.
This is very useful to dynamically apply a row style, depending on the data.
See rowStyle sample provided.

adrianbadea committed:
Add a sample for the rowStyle.

adrianbadea committed:
For invalid dataIndex, getColumn() returns the last column in grid

If no column is available for the 'dataIndex' argument, then the
'getColumn()' function returns the last column in the grid. It should
return 'null' instead.

mitchellsimoens commented on Ext.ux.touch.grid/List.js:
```diff
@@ -219,22 +241,32 @@ Ext.define('Ext.ux.touch.grid.List', {
 
         for (; c < cNum; c++) {
-            column = columns[c];
+            var column = columns[c];
```
The variable should be defined before the for loop like it was

mitchellsimoens commented on rowStyle.js:
```diff
@@ -0,0 +1,154 @@
+             	// rowStyle : 'background-color: #D1D1D1;',
+
+        		rowStyle : function(data) {
+					if (data)
```
Code is not formatted and indented correctly

mitchellsimoens commented on rowStyle.js:
```diff
@@ -0,0 +1,154 @@
+            }
+		});
+
+        var store = Ext.create('Ext.data.Store', {
```
This store is not indented properly

mitchellsimoens commented on rowStyle.js:
```diff
@@ -0,0 +1,154 @@
+
+        		rowStyle : function(data) {
+					if (data)
+					{
```
bracket should be on same line as the if

mitchellsimoens commented on Ext.ux.touch.grid/List.js:
```diff
@@ -219,22 +241,32 @@ Ext.define('Ext.ux.touch.grid.List', {
 
+    getRowStyle : function (data) {
+        var me = this,
+            rowStyle = me._rowStyle;
```
equal signs not lined up

mitchellsimoens commented on Ext.ux.touch.grid/List.js:
```diff
@@ -196,10 +204,24 @@ Ext.define('Ext.ux.touch.grid.List', {
+        		rstl = null;
+        	
+        	if (Ext.isFunction(rowCls) || Ext.isString(rowCls)) {
+            	renderers._getRowCls = Ext.bind(me.getRowCls, me);
```
Instead of doing an Ext.bind each time you could do this in the constructor before a callParent call

adrianbadea replied:
Maybe I'm missing something, but Ext.bind gets called once only, when the row template is created.

mitchellsimoens commented on Ext.ux.touch.grid/List.js:
```diff
@@ -196,10 +204,24 @@ Ext.define('Ext.ux.touch.grid.List', {
+        	}
+        
+        	if (Ext.isFunction(rowStyle) || Ext.isString(rowStyle)) {
+            	renderers._getRowStyle = Ext.bind(me.getRowStyle, me);
```
Instead of doing an Ext.bind each time you could do this in the constructor before a callParent call

adrianbadea replied:
Maybe I'm missing something, but Ext.bind gets called once only, when the row template is created.

adrianbadea committed:
Re-formatting

adrianbadea committed:
Update to work with 2.1.0 final

adrianbadea committed:
Define 'column' variable before the for loop like it was

mitchellsimoens commented:
If you show/hide a column on the fly, it will regenerate the template so the Ext.bind would be executed multiple times.

adrianbadea committed:
Minimize the calls for Ext.bind

Instead of calling the Ext.bind each time a row template is built,
create the function once and save its reference.

adrianbadea commented:
I've updated all the files, as per your suggestions. Please review them and let me know if anything else. Thanks.

adrianbadea committed:
Issue #31

Ensure the item template gets applied when the grid structure changes:
toggle columns.

adrianbadea committed:
Synch with master

adrianbadea commented:
Hi Mitchell,

When you have a moment, please review and merge the changes for the row style feature and issue #31. Thanks.
